src/components/input/inputFileAvatar/inputFileAvatar.tsx

```diff
@@ -0,0 +1,184 @@
+                    maxDimension > 0 && (image.width > maxDimension || image.height > maxDimension);
+                if (onlySquare && image.height !== image.width) {
+                    setCriticalSelectError(ERROR_MESSAGES.ONLY_SQUARE);
+                    setSelectState(SelectState.ERROR);
+                } else if (isBelowMinDimension || isAboveMaxDimension) {
+                    setCriticalSelectError(ERROR_MESSAGES.WRONG_DIMENSION);
+                    setSelectState(SelectState.ERROR);
+                } else {
+                    setImagePreview(image.src);
+                    setSelectState(SelectState.SUCCESS);
+                    setSelectError(undefined);
+                    onFileSelect?.(file);
+                }
+            };
+
+            image.addEventListener('load', onImageLoad);
+            image.addEventListener('error', () => {
+                setCriticalSelectError(ERROR_MESSAGES.FAILED);
+                setSelectState(SelectState.ERROR);
+            });
+            image.src = URL.createObjectURL(file);
+        },
+        [
+            ERROR_MESSAGES.FAILED,
+            ERROR_MESSAGES.FILE_TYPE,
+            ERROR_MESSAGES.QUANTITY,
+            ERROR_MESSAGES.TOO_LARGE,
+            ERROR_MESSAGES.WRONG_DIMENSION,
+            ERROR_MESSAGES.ONLY_SQUARE,
+            maxDimension,
+            minDimension,
+            onFileSelect,
+            onlySquare,
+        ],
+    );
+
+    const { getRootProps, getInputProps } = useDropzone({
+        accept: ALLOWED_FILE_TYPES,
+        ...(maxFileSize > 0 && { maxSize: MAX_FILE_SIZE }),
+        disabled: isDisabled ?? selectState === SelectState.SELECTING,
+        onDrop,
+    });
+
+    const handleCancel = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
+        setImagePreview(null);
+        setSelectState(SelectState.IDLE);
+        if (imagePreview) {
+            URL.revokeObjectURL(imagePreview);
+        }
+    };
+
+    const selecterVariantStateClasses = classNames(
+        'group flex size-16 items-center justify-center rounded-full bg-neutral-0 hover:shadow-neutral focus:outline-none focus-visible:ring  focus-visible:ring-offset', // Default
+        {
+            ['border-[1px] border-neutral-100 hover:border-neutral-200 border-dashed cursor-pointer focus-visible:ring-primary']:
+                selectState === SelectState.IDLE, // IDLE State
+            ['cursor-default border-[1px] border-primary-400 focus-visible:ring-primary']:
+                selectState === SelectState.SELECTING, // SELECTING State
+            ['cursor-default border-[5px] border-neutral-100 border-dashed focus-visible:ring-primary']:
+                selectState === SelectState.SUCCESS, // SUCCESS State
+            ['border-[1px] border-critical-500 hover:border-critical-600 border-dashed cursor-pointer focus-visible:ring-critical']:
+                selectState === SelectState.ERROR, // ERROR State
```

Comment:
This can be a constant defined outside the component like `stateToClassNames`. Check how we are currently doing this on other components